.github/workflows/ci.yml

```diff
@@ -43,40 +43,53 @@ jobs:
       - name: basedpyright
         run: uv run poe pyright
 
-  typetest:
-    timeout-minutes: 5
+  generate-matrix:
+    name: Generate Test Matrix
     runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
       - uses: actions/checkout@v4
 
-      - name: setup uv
-        uses: astral-sh/setup-uv@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
 
-      - name: install
-        run: uv sync --frozen --python 3.13
-
-      - name: typetest
-        run: uv run poe typetest
+      - name: Install Python Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install scipy
```

Comment:
the `scipy` version should probably be pinned to the one supported by `scipy-stubs`